include/svs/leanvec/leanvec_concept.h

```diff
@@ -0,0 +1,536 @@
+/*
+ * Copyright 2023 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "svs/quantization/lvq/lvq_concept.h"
+
```

Comment:
I remember we discussed it earlier, but the file names: "lvq_concept.h" or "leanvec_concept.h" don't make sense now. Why not we combine them with "*lvq/leanvec_common.h"? If not possible, then we can rename something like lvq/leanvec_loader.h

Reply:
They are separated because lvq_common defines functionality that lvq depends on, and lvq_concept depends on lvq (see graphic in private PR for further context). It's possible these could be combined into a single file with some forward declaration if we want to, but not sure how feasible this would be.

I can rename the file from concept to loader.